pkg/batcher/batcher.go

```diff
@@ -0,0 +1,180 @@
+package batcher
+
+import (
+	"fmt"
+	"github.com/StackVista/stackstate-agent/pkg/collector/check"
+	"github.com/StackVista/stackstate-agent/pkg/serializer"
+	"github.com/StackVista/stackstate-agent/pkg/topology"
+	"sync"
+)
+
+var (
+	batcherInstance Batcher
+	batcherInit     sync.Once
+)
+
+// Batcher interface can receive data for sending to the intake and will accumulate the data in batches. This does
+// not work on a fixed schedule like the aggregator but flushes either when data exceeds a threshold, when
+// data is complete.
+type Batcher interface {
+	SubmitComponent(checkID check.ID, instance topology.Instance, component topology.Component)
+	SubmitRelation(checkID check.ID, instance topology.Instance, relation topology.Relation)
+	SubmitStartSnapshot(checkID check.ID, instance topology.Instance)
+	SubmitStopSnapshot(checkID check.ID, instance topology.Instance)
+	SubmitComplete(checkID check.ID)
+	Shutdown()
+}
+
+// InitBatcher initializes the global batcher instance
+func InitBatcher(serializer serializer.AgentV1Serializer, hostname, agentName string, batchLimit int) {
+	batcherInit.Do(func() {
+		batcherInstance = newAsynchronousBatcher(serializer, hostname, agentName, batchLimit)
+	})
+}
+
+func newAsynchronousBatcher(serializer serializer.AgentV1Serializer, hostname, agentName string, batchLimit int) AsynchronousBatcher {
+	batcher := AsynchronousBatcher{
+		builder:    NewTopologyBuilder(batchLimit),
```

Comment:
rename `batchLimit` according to the name we chosen in the TopologyBuilder

Reply:
done